rusty_pizza_server/src/order_model/order.rs

```diff
@@ -0,0 +1,97 @@
+use crate::order_model::meals::Meals;
+use crate::order_model::user::User;
+use crate::util::money::Money;
+use std::collections::HashMap;
+use std::fmt::{self, Display, Formatter};
+use std::rc::Rc;
+
+#[derive(Debug, PartialEq)]
+enum OrderStatus {
+    Open,
+    Ordering,
+    Ordered(String),
+    Delivered,
+}
+
+impl fmt::Display for OrderStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Order {
+    meals: HashMap<Rc<User>, Meals>,
+    status: OrderStatus,
+    manager: Rc<User>,
+}
+
+impl Order {
+    pub fn new(manager: Rc<User>) -> Order {
+        Order {
+            meals: HashMap::new(),
+            status: OrderStatus::Open,
+            manager,
+        }
+    }
+
+    pub fn add_user(&mut self, user: Rc<User>) {
+        let meals = Meals::new(user.clone());
+        self.meals.insert(user, meals);
+    }
+
+    pub fn get_order_status(self) -> String {
+        self.status.to_string()
```

Comment:
Why formatting this out as a string?

Reply:
This is a really good question. I think i had an idea during modularize, but at the end this method is not needed anymore.
I will remove it